tests/FakeItEasy.Specs/ValueTaskReturnValueSpecs.cs

```diff
@@ -0,0 +1,166 @@
+namespace FakeItEasy.Specs
+{
+    using System;
+    using System.Threading.Tasks;
+    using FluentAssertions;
+    using Xbehave;
+
+    public class ValueTaskReturnValueSpecs
+    {
+        public interface IReturnValueTask
+        {
+            ValueTask<int> GetValueAsync();
+
+            ValueTask<int> GetValueAsync(int a);
+
+            ValueTask<int> GetValueAsync(int a, string b);
+
+            ValueTask<int> GetValueAsync(int a, string b, bool c);
+        }
+
+        [Scenario]
+        public void ValueTaskSpecificValue(IReturnValueTask fake, ValueTask<int> task)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<IReturnValueTask>());
+
+            "And a ValueTask-returning method of the fake configured to return a specific value"
+                .x(() => A.CallTo(() => fake.GetValueAsync()).Returns(42));
+
+            "When the configured method is called"
+                .x(() => task = fake.GetValueAsync());
+
+            "Then it returns a successfully completed ValueTask"
+                .x(() => task.IsCompletedSuccessfully.Should().BeTrue());
+
+            "And the ValueTask's result is the configured value"
+                .x(() => task.Result.Should().Be(42));
+        }
+
+        [Scenario]
+        public void ValueTaskLazilyComputed(IReturnValueTask fake, Func<int> valueProducer, ValueTask<int> task)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<IReturnValueTask>());
+
+            "And a value producer configured to return a specific value"
+                .x(() => valueProducer = A.Fake<Func<int>>(o => o.ConfigureFake(f => A.CallTo(() => f()).Returns(42))));

```

Comment:
This is to be able to assert that the value producer was called. If you think it's overkill, I'll simplify it.